build.rs

```diff
@@ -63,7 +171,17 @@ fn extract_tar_gz(path: &Path, output_dir: &Path) {
 
 /// Determines whether the CRT is being statically or dynamically linked.
 fn static_crt() -> bool {
-    var("CARGO_ENCODED_RUSTFLAGS")
-        .unwrap_or_default()
-        .contains("target-feature=+crt-static")
-}
\ No newline at end of file
+    env::var("CARGO_ENCODED_RUSTFLAGS")
+        .map(|flags| flags.contains("target-feature=+crt-static"))
+        .unwrap_or(false)
+}
+
+/// Get global cache path for downloaded file.
+fn get_cached_path() -> PathBuf {
+    const CACHE_FOLDER_NAME: &str = "mach_dxcompiler_rs";
+    if let Some(dir_str) = env::var_os("RUST_CACHED_PATH_ROOT") {
+        PathBuf::from(dir_str).join(CACHE_FOLDER_NAME)
+    } else {
+        env::temp_dir().join(CACHE_FOLDER_NAME)
```

Comment:
Would it make more sense to use `env::var("OUT_DIR").join(CACHE_FOLDER_NAME)` for the default cache directory? This way if the user runs `cargo clean`, the cached binaries go away (whereas they may pile up indefinitely in the temp directory).

Reply:
Further, what happens if the user downloads this crate as a dependency, builds the crate (which creates a temp folder and extracts the `.lib` there), and then clears their temp directory? Wouldn't Cargo get confused, because the build script would not rerun and Cargo would still be looking for the `.lib` in the old temp directory?

Reply:
That makes sense.